src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs

```diff
@@ -187,7 +195,7 @@ private void DisposeIndexWriter(bool waitForMerges = true)
 
             try
             {
-                writer.Dispose(waitForMerges);
+                writer.Dispose(waitForMerges: false);
```

Comment:
Did you change the default behavior?

Reply:
waitingForMerges: true triggers a new index merge (if needed)
We are already doing the merge during the commit stage.

Reply:
It could be a leftover from 3.5 when we had a concurrent merger, now we have a serial one?

Reply:
very likely, this also had the issue of the deadlock